src/package-install-scripts.js

```diff
@@ -280,15 +292,55 @@ export default class PackageInstallScripts {
 
     await Promise.all(workers);
 
-    // cache all build artifacts
-    for (const pkg of pkgs) {
-      if (this.packageCanBeInstalled(pkg)) {
-        const ref = pkg._reference;
-        invariant(ref, 'expected reference');
-        const loc = this.config.generateHardModulePath(ref);
-        const beforeFiles = beforeFilesMap.get(loc);
-        invariant(beforeFiles, 'files before installation should always be recorded');
-        await this.saveBuildArtifacts(loc, pkg, beforeFiles, set.spinners[0]);
+    // generate built package as prebuilt one for offline mirror
+    if (this.config.packBuiltPackages) {
+      for (const pkg of pkgs) {
+        if (this.packageCanBeInstalled(pkg)) {
+          const offlineMirrorPath = this.config.getOfflineMirrorPath();
+          if (!offlineMirrorPath) {
+            break;
+          }
+          let prebuiltPath = path.join(offlineMirrorPath, 'prebuilt');
+          if (!await fs.exists(prebuiltPath)) {
+            await fs.mkdirp(prebuiltPath);
+          }
+          const prebuiltFilename = getPlatformSpecificPackageFilename(pkg);
+          prebuiltPath = path.join(prebuiltPath, prebuiltFilename + '.tgz');
+          const ref = pkg._reference;
+          invariant(ref, 'expected reference');
+          const loc = this.config.generateHardModulePath(ref);
+          const pkgConfig = await Config.create(
+            {
+              cwd: loc,
+            },
+            this.reporter,
+          );
+          const stream = await pack(pkgConfig, loc);
```

Comment:
Does `pack` also includes the build artifacts? :o

Reply:
I pass `loc` which from local `node_modules` with built artifacts, pack just generates a .tgz from a folder

Reply:
I'll name loc better to be clear